### DISCUSSION TITLE
Astro projects using MongoDB fail to build

### DISCUSSION THREAD

**Discussion by seeplusplus** [vercel/community#3541] [Help] [answered]
### Summary

Astro projects seem to fail any build where there's any reference to the mongodb npm package in code. Builds succeed if the package is in the package.json but fail as soon as any code references it.

### Example

https://github.com/seeplusplus/minimal-example

First commit where build failures can be seen:

https://github.com/seeplusplus/minimal-example/commit/3663c02a64a099b627df9c601e52e03e1c088604

I thought maybe this was because I was using `import` syntax when mongodb docs use `require`, but that didn't work for me either as seen by the following commit:

https://github.com/seeplusplus/minimal-example/commit/d062182ff4b37d148c370513a6adf99f460e17c2

### Steps to Reproduce

1. Start an Astro project in vercel using the [Astro boilerplate](https://vercel.com/new/templates/astro/astro-boilerplate)
2. `pnpm i mongodb`
3. Add
```
import { MongoClient } from 'mongodb';
console.log(MongoClient);
```
to `index.astro` (or any JS that will get invoked on build).

4. `git push`
4. Build will fail in Vercel due to `Could not resolve "os"` and in fact tons of other `Could not resolve .+` errors.

**Comment by koslowdavida** [comment]
Getting this as well; bumping for visibility

**Comment by amyegan** [comment, marked as answer]
Hey there. This type of error generally means that the code is trying to do something on the client-side that needs to be done on the server-side.

**Reply by seeplusplus** [reply]
~~Any thoughts on why that may be? In the example all references to MongoDB were done above the `---` in `.astro` files; my understanding is that this should be only executed server-side.~~

Edit: just retested this locally and got build failures. Apologies, I thought I had double checked this before opening this discussion. Closing this discussion and moving to the Astro support channel. Thank you for your response!

**Reply by seeplusplus** [reply]
@amyegan I wanted to follow up; I looked into this some more this afternoon. It turns out, I don't get this error when I start an Astro project from scratch (using `pnpm create astro && pnpm astro add vercel` instead of the template). It looks like this is caused by using the `@astrojs/vercel/edge` adapter instead of the `@astrojs/vercel/serverless` adapter.
